core/admin/mailu/configuration.py

```diff
@@ -123,6 +123,9 @@ def init_app(self, app):
 
         self.config['RATELIMIT_STORAGE_URL'] = 'redis://{0}/2'.format(self.config['REDIS_ADDRESS'])
         self.config['QUOTA_STORAGE_URL'] = 'redis://{0}/1'.format(self.config['REDIS_ADDRESS'])
+        self.config['SESSION_COOKIE_SAMESITE'] = 'Strict'
+        self.config['SESSION_COOKIE_HTTPONLY'] = True
+        self.config['SESSION_COOKIE_SECURE'] = self.config['TLS_FLAVOR'] != 'notls'
```

Comment:
I don't think that check is correct. For example if someone uses mail or mail-letsencrypt they don't necessarily access admin via https. The other way around when someone uses cert or letsencrypt they could still access admin unencrypted by exposing it directly (iirc setup does this by default).

Not sure what the proper solution would be. Adding x-forwarded-proto to nginx and checking for that? But that has other caveats, e.g. we'd have to check if the request originated from nginx.

Reply:
I guess this is related to #1740
I'm not sure what the proper solution is either.

Maybe we should go further and delegate all of this to something like flask-talisman ?

Reply:
One of the complications is that we don't have a flask request context at configuration time...

I'm not convinced that auto-detecting the right value at runtime is the way forward (hard to cover all the usecases, will be difficult to debug by users, ...). What about just documenting it and allowing the default to be overridden in the configuration?

Reply:
Meh. The complexity for handling this is just too high.

https://mailu.io/1.7/general.html#philosophy
"Mailu has a central front container that routes all HTTP and mail traffic."

Can't we just stick with that? :)
And if we do, would you rather I get nginx to set the cookie flags? (set_cookie_flag https://docs.nginx.com/nginx/admin-guide/dynamic-modules/cookie-flag/)

Reply:
I've implemented what was agreed on matrix in 0dcc059